Source/DafnyCore/AST/Expressions.cs

```diff
@@ -2515,6 +2458,10 @@ public class LetOrFailExpr : ConcreteSyntaxExpression {
     Rhs = rhs;
     Body = body;
   }
+
+  // TODO: Children should take into account Rhs and Body when we support pre-resolved children
```

Comment:
TODO are always for a future PR. I prefer to keep it here because I'm introducing the code that seems wrong and this TODO explain what will to be done so that it does not feel wrong anymore in the future.

Reply:
> TODO are always for a future PR

I use them within a single PR. I try to remove them before the review but I sometimes I forget.

> I prefer to keep it here because I'm introducing the code that seems wrong

I think right now `Children` is mostly returning resolved children, and that's also what the documentation says it should do, so not having that right now looks fine to me.

Reply:
> that's also what the documentation says it should do, so not having that right now looks fine to me.

I disagree that we remove this TODO because I think the documentation is misleading, but I'll do it anyway because there are more important stuff to discuss.